opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -46,95 +62,179 @@ def __init__(
 
     def __init__(
         self,
+        metric_exporters: Sequence[MetricExporter] = (),
+        metric_readers: Sequence[MetricReader] = (),
+        views: Sequence[View] = (),
         resource: Resource = Resource.create({}),
         shutdown_on_exit: bool = True,
+        use_always_matching_view: bool = True,
     ):
-        self._resource = resource
+        self._lock = Lock()
         self._atexit_handler = None
+        self.__use_always_matching_view = use_always_matching_view
 
         if shutdown_on_exit:
             self._atexit_handler = register(self.shutdown)
 
-        self._metric_readers = []
-        self._metric_exporters = []
-        self._views = []
-        self._shutdown = False
+        self.__metric_readers = metric_readers
 
-    def get_meter(
-        self,
-        name: str,
-        version: Optional[str] = None,
-        schema_url: Optional[str] = None,
-    ) -> Meter:
+        for metric_reader in self._metric_readers:
+            metric_reader._register_meter_provider(self)
 
-        if self._shutdown:
-            _logger.warning(
-                "A shutdown `MeterProvider` can not provide a `Meter`"
-            )
-            return _DefaultMeter(name, version=version, schema_url=schema_url)
+        self.__metric_exporters = metric_exporters
 
-        return Meter(InstrumentationInfo(name, version, schema_url), self)
+        self.__views = views
 
-    def shutdown(self):
-        # FIXME implement a timeout
+        if self.__use_always_matching_view:
+            self.__views = [*self.__views, View(instrument_name="*")]
 
-        if self._shutdown:
-            _logger.warning("shutdown can only be called once")
-            return False
+        self.__resource = resource
+        self._shutdown = False
+        self.__synchronous_instruments = []
+        self.__asynchronous_instruments = []
 
-        result = True
+    @property
+    def _metric_readers(self):
+        return self.__metric_readers
 
-        for metric_reader in self._metric_readers:
-            result = result and metric_reader.shutdown()
+    @property
+    def _use_always_matching_view(self):
+        return self.__use_always_matching_view
 
-        for metric_exporter in self._metric_exporters:
-            result = result and metric_exporter.shutdown()
+    @property
+    def _resource(self):
+        return self.__resource
 
-        self._shutdown = True
+    @property
+    def _metric_exporters(self):
+        return self.__metric_exporters
 
-        if self._atexit_handler is not None:
-            unregister(self._atexit_handler)
-            self._atexit_handler = None
+    @property
+    def _views(self):
+        return self.__views
 
-        return result
+    @property
+    def _synchronous_instruments(self):
+        return self.__synchronous_instruments
+
+    @property
+    def _asynchronous_instruments(self):
+        return self.__asynchronous_instruments
+
+    def _create_instrument(
```

Comment:
This doesn't look much cleaner than just calling the instrument constructor in each method create method individually. You even have to do an `isinstance()` check this way, which isn't really necessary

Reply:
Sure, we can also implement this without `_create_instrument` if you prefer, fixed.